src/mpid/ch4/netmod/ofi/ofi_coll.h

```diff
@@ -12,20 +12,52 @@
 #define OFI_COLL_H_INCLUDED
 
 #include "ofi_impl.h"
+#include "ch4_coll_select.h"
+#include "ch4_coll_params.h"
+#include "ofi_coll_select.h"
 
 #undef FUNCNAME
 #define FUNCNAME MPIDI_NM_mpi_barrier
 #undef FCNAME
 #define FCNAME MPL_QUOTE(FUNCNAME)
-static inline int MPIDI_NM_mpi_barrier(MPIR_Comm * comm_ptr, MPIR_Errflag_t * errflag)
+static inline int MPIDI_NM_mpi_barrier(MPIR_Comm * comm_ptr, MPIR_Errflag_t * errflag,
+                                       void * ch4_algo_parameters_container_in)
```

Comment:
MPIDI_NM_mpi_barrier is the general netmod function which have its own implementations for each particular netmod: OFI, UCX, etc. To hide the details about particular netmod in the general NM functions we pass void * as a parameter. For netmod specific functions like MPIDI_OFI_Barrier_recursive_doubling we pass  MPIDI_OFI_coll_algo_container_t * since it is clear from the prefix "MPIDI_OFI" of the name of the function which type of te container shoul be used. The same idea was used for SHM/POSIX: in shm_coll.h we pass void * as a parameter for MPIDI_SHM_mpi_barrier function (since it is general), but in posix_coll.h we pass MPIDI_POSIX_coll_algo_container_t * for MPIDI_POSIX_mpi_barrier. 
The difference between NM and SHM is that NM doesn't have separate nm_coll.h with general for all netmods functions while SHM has.